model/model/v1alpha/model.proto

```diff
@@ -1737,3 +1738,117 @@ message UndeployOrganizationModelAdminRequest {
 
 // UndeployOrganizationModelAdminResponse represents a response for a undeployed model
 message UndeployOrganizationModelAdminResponse {}
+
+// ModelTrigger contains information about trigger of models.
+message ModelTrigger {
+  // TriggerStatus defines the status of a model trigger
+  enum TriggerStatus {
+    // Unspecified
+    TRIGGER_STATUS_UNSPECIFIED = 0;
+    // model trigger in progress
+    TRIGGER_STATUS_PROCESSING = 1;
+    // model trigger succeeded
+    TRIGGER_STATUS_COMPLETED = 2;
+    // model trigger failed
+    TRIGGER_STATUS_FAILED = 3;
+    // model trigger is waiting to be executed
+    TRIGGER_STATUS_QUEUED = 4;
+  }
+
+  // TriggerSource defines the source of a model trigger
+  enum TriggerSource {
+    // Unspecified
+    TRIGGER_SOURCE_UNSPECIFIED = 0;
+    // model trigger from frontend UI
+    TRIGGER_SOURCE_CONSOLE = 1;
+    // model trigger from API or SDK
+    TRIGGER_SOURCE_API = 2;
+  }
+
+  // Model Trigger UUID.
+  string uid = 1 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model UUID.
+  string model_uid = 2 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Pipeline Trigger UUID.
+  string trigger_uid = 3 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model trigger status
+  TriggerStatus status = 4 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Model visibility.
+  Model.Visibility visibility = 5 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Trigger source
+  TriggerSource source = 6 [(google.api.field_behavior) = OUTPUT_ONLY];
+  // Trigger start time
+  google.protobuf.Timestamp start_time = 7 [(google.api.field_behavior) = OUTPUT_ONLY];
```

Comment:
Are `start_time` and `create_time` redundant?

Reply:
I added this because some model infer requests might wait in queue for a while before actually starting and `create_tme` would be earlier than `start_time`...
As we cannot get the actual `start_time` right now, I'll delete `start_time` and we can add it later

Reply:
It makes sense, definitely in the database, although I don't know if exposing that publicly adds clarity or confusion. In either case I'd document the fields that are exposed to be clear about what they mean (request time vs actual execution time)